protocol/contracts/oracle/src/stub/alarms.rs

```diff
@@ -1,50 +1,80 @@
 use oracle_platform::OracleRef;
+
+use currency::{Currency, Group};
 use platform::batch::Batch;
 use sdk::cosmwasm_std::{wasm_execute, Addr};
 
-use crate::api::{
-    alarms::{Alarm, AlarmCurrencies, Error, ExecuteMsg, Result},
-    BaseCurrencies,
-};
+use crate::api::alarms::{Alarm, Error, ExecuteMsg, Result};
 
-pub trait PriceAlarms
+pub trait PriceAlarms<AlarmCurrencies, OracleBaseG>
```

Comment:
The 'OracleBaseG` is better to be provided as an associative type than a type parameter because it is a function of Self, the implementer.
I realize this would require a type parameter of OracleStub, which would bring many changes, though.